tsdb/head.go

```diff
@@ -293,6 +293,21 @@ func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int
 		tombstones: tombstones.NewMemTombstones(),
 		iso:        newIsolation(),
 		deleted:    map[uint64]int{},
+		appendPool: sync.Pool{
+			New: func() interface{} {
+				return make([]record.RefSample, 0, 512)
+			},
+		},
+		seriesPool: sync.Pool{
+			New: func() interface{} {
+				return make([]*memSeries, 0, 512)
+			},
+		},
+		bytesPool: sync.Pool{
+			New: func() interface{} {
+				return make([]byte, 0, 8192)
```

Comment:
Still not convinced with your explanation for increasing this size, do you have any benchmark results? This pool seems to be used only during logging into the WAL, so that data doesn't increase with time. And it should be fine to have a slightly smaller size here, if it grows at some point, it will be re-used, hence there won't be frequent allocations.

Reply:
1. `buf`'s cap is no longer the same
After taking out the `buf` from the pool, if the `buf`'s cap changes, put this `buf`back in the pool, and then take it out, the `buf`'s cap is equal to last changed value. Over time, the `buf` in the pool will no longer be the same size. Small caps are more susceptible.

Reply:
2. benchmark
Maybe there is no optimal size. Because there are more situations.

Reply:
Is it appropriate to introduce metric here to record(considering cost)？I have no idea about it.

Reply:
Without any real numbers on if it is actually benefiting much and not affecting the memory of smaller setups, I would not like to change the size here. The growth in the slice cap would be only a few times before it becomes stable and that is fine.

Reply:
good